File: 3600-find-the-k-th-character-in-string-game-i/find-the-k-th-character-in-string-game-i.py
class Solution:
    def kthCharacter(self, k: int) -> str:
        # Trả về ký tự thứ k trong chuỗi sinh ra, dựa trên số bit 1 trong (k-1)
        return chr(ord('a') + bin(k - 1).count('1'))

File: 3600-find-the-k-th-character-in-string-game-i/find-the-k-th-character-in-string-game-i.java
class Solution {
    public char kthCharacter(int k) {
        // Bản chất của bài toán: đếm số bit 1 trong (k-1)
        // Sau mỗi lần nối chuỗi, chữ cái được chuyển sang ký tự kế tiếp trong bảng chữ cái
        // Vị trí k sẽ có ký tự = 'a' + số lần ký tự được "tăng" -> tương đương với số bit 1 trong k-1
        return (char) ('a' + Integer.bitCount(k - 1));
    }
}